### PR TITLE
Assignment 08/11

### PR DESCRIPTION
**Pull request by sansidia**

### CODE REVIEW

File: userdata.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    printf("Welcome to Food Thingies!\n");
+    printf("Please sign in to continue!\n");
+    printf("---Username:\n");
+    gets(username);
+    printf("---Password:\n");
+    gets(password);
+}
+
+void printForm(char username[], char food[], int foodPrice, char drink[], int drinkPrice, int drinkCode, int noDrink, int cutleryChoice, char additionalInfo[]) {
+    printf("-----------------\n"
```
Comment (Ranapop) [inline userdata.c]: a bit too long, you could call some functions in here

File: main.c [hunk 2]
```diff
@@ -1,202 +1,49 @@
...
-    int foodSubtypeChoice = -1;
-    char drinkOptions[NO_DRINKS][20] = {"Coca-Cola", "Fanta", "Lipton", "Water"};
-    int drinkOptionPrices[NO_DRINKS] = {5, 5, 5, 4};
-    int drinkChoice = -1;
-    int cutleryChoice = -1;
-    char additionalInfo[100];
-    char choice;
-    int state = 0;
-    int isOrderConfirmed = 0;
+    char username[CHAR_LENGTH], password[CHAR_LENGTH], foodTypes[NO_FOOD_TYPE][CHAR_LENGTH] = {"Pizza", "Pasta", "Salad"}, foodSubtypes[NO_FOOD_TYPE][NO_FOOD_SUBTYPE][CHAR_LENGTH] = {{"Pizza con Pollo", "Pizza Diavola", "Pizza Margherita"}, {"Chicken alfredo", "Mac&cheese"}, {"Tuna Salad", "Chicken Salad", "Greek Salad", "Cobb"}}, drinkOptions[NO_DRINKS][CHAR_LENGTH] = {"Coca-Cola", "Fanta", "Lipton", "Water"}, cutleryOptions[2][5] = {"Yes", "No"}, additionalInfo[100];
```
Comment (Ranapop) [inline main.c]: don't do this... please format these lines properly...

**Comment by sansidia** [comment]
I made the changes, hope all is good now.

File: main.c [hunk 3]
```diff
@@ -1,202 +1,51 @@
...
     int noFoodSubtypes[NO_FOOD_TYPE] = {3, 2, 4};
-    char foodSubtypes[NO_FOOD_TYPE][NO_FOOD_SUBTYPE][20] = {
-            {"Pizza con Pollo", "Pizza Diavola", "Pizza Margherita"},
-            {"Chicken alfredo", "Mac&cheese"},
-            {"Tuna Salad", "Chicken Salad", "Greek Salad", "Cobb"}
-    };
-    int foodSubtypeChoice = -1;
-    char drinkOptions[NO_DRINKS][20] = {"Coca-Cola", "Fanta", "Lipton", "Water"};
+    char foodSubtypes[NO_FOOD_TYPE][NO_FOOD_SUBTYPE][CHAR_LENGTH] = {{"Pizza con Pollo", "Pizza Diavola", "Pizza Margherita"},{"Chicken alfredo", "Mac&cheese"},{"Tuna Salad", "Chicken Salad", "Greek Salad", "Cobb"} };
+    char drinkOptions[NO_DRINKS][CHAR_LENGTH] = {"Coca-Cola", "Fanta", "Lipton", "Water"};
```
Comment (Ranapop) [inline main.c]: again, please format them properly... I would say such long one liners are unacceptable...

File: main.c [hunk 4]
```diff
@@ -1,202 +1,51 @@
...
-    int foodTypeChoice = -1;
-    int foodSubtypePrices[NO_FOOD_TYPE][4] = {
-            {21, 23, 19},
-            {23, 21},
-            {23, 22, 19, 21}
-    };
+    char username[CHAR_LENGTH], password[CHAR_LENGTH];
+    char foodTypes[NO_FOOD_TYPE][CHAR_LENGTH] = {"Pizza", "Pasta", "Salad"};
+    int foodTypeChoice = -1, foodSubtypeChoice = -1, drinkChoice = -2, cutleryChoice = -1, choice;
+    int foodSubtypePrices[NO_FOOD_TYPE][NO_FOOD_SUBTYPE] = {{21, 23, 19},{23, 21},{23, 22, 19, 21} };
```
Comment (Ranapop) [inline main.c]: do not define on one line

**Comment by sansidia** [comment]
I might get home very late so I can't add the CMakeLists file. I modified the variable declarations.

**Comment by sansidia** [comment]
added CMakeLists.txt

**Pull request merged**